libraries/rush-lib/src/api/RushConfigurationProject.ts

```diff
@@ -298,17 +298,30 @@ export class RushConfigurationProject {
       ]) {
         if (dependencySet) {
           for (const [dependency, version] of Object.entries(dependencySet)) {
+            const dependencySpecifier: DependencySpecifier | undefined = tryParseDependencySpecifier(
+              dependency,
+              version
+            );
+            if (!dependencySpecifier) {
+              // Don't recognize the format. Ignore.
+              continue;
```

Comment:
Will this then give an error?

Reply:
Rush won't complain, but your package manager might.

Reply:
My main concern about trying to surface an error here is that we would want to surface every dependency Rush doesn't understand at once, and we don't currently have a system for that.

Reply:
> Rush won't complain, but your package manager might.

From a support perspective, we probably would prefer to block protocol specifiers that the Rush maintainers have not explicitly tested and agreed to support.  A permissive policy would have a couple liabilities:
- If PNPM reports an error, it can sometimes be a confusing experience because PNPM assumes it was not wrapped by Rush and that its workspace was not relocated to `common/temp`
- If PNPM does NOT report an error, the monorepo can nonetheless end up in a situation where installs are broken (due to incompatibility with Rush features) or not reliable for a professional monorepo (e.g. `file:` or `git:` specifiers)